go-controller/pkg/cni/helper_linux.go

```diff
@@ -338,6 +338,43 @@ func (pr *PodRequest) ConfigureInterface(namespace string, podName string, ifInf
 	return []*current.Interface{hostIface, contIface}, nil
 }
 
+func (pr *PodRequest) deletePodConntrack() {
+	if pr.CNIConf.PrevResult == nil {
+		return
+	}
+	result, err := current.NewResultFromResult(pr.CNIConf.PrevResult)
+	if err != nil {
+		klog.Warningf("could not convert result to current version: %v", err)
+		return
+	}
+
+	for _, ip := range result.IPs {
+		// Skip known non-sandbox interfaces
+		if ip.Interface != nil {
+			intIdx := *ip.Interface
+			if intIdx >= 0 &&
+				intIdx < len(result.Interfaces) && result.Interfaces[intIdx].Sandbox == "" {
+				continue
+			}
+		}
+		if ip.Version == "4" {
+			filterTypes := []netlink.ConntrackFilterType{netlink.ConntrackOrigSrcIP, netlink.ConntrackOrigDstIP, netlink.ConntrackReplyAnyIP}
+			for _, filterType := range filterTypes {
+				var filter *netlink.ConntrackFilter
+				err := filter.AddIP(filterType, ip.Address.IP)
+				if err != nil {
+					klog.Warningf("could not add IP: %s to conntrack filter: %v", ip.Address.IP.String(), err)
+				}
+				_, err = netlink.ConntrackDeleteFilter(netlink.ConntrackTable, netlink.UDP_PROTO, filter)
```

Comment:
@JacobTanenbaum why only UDP, and why. not TCP?

Reply:
@girishmg I don't know the exact details, but I believe because this is what kube-proxy does. I never clears conntrack for TCP, only for UDP. There are multiple places that it does this:

```
// After a UDP endpoint has been removed, we must flush any pending conntrack entries to it, or else we
// risk sending more traffic to it, all of which will be lost (because UDP).
func (proxier *Proxier) deleteEndpointConnections(connectionMap []proxy.ServiceEndpoint) {
```

```
					if lp.Protocol == "udp" {
						// TODO: We might have multiple services using the same port, and this will clear conntrack for all of them.
						// This is very low impact. The NodePort range is intentionally obscure, and unlikely to actually collide with real Services.
						// This only affects UDP connections, which are not common.
						// See issue: https://github.com/kubernetes/kubernetes/issues/49881
						err := conntrack.ClearEntriesForPort(proxier.exec, lp.Port, isIPv6, v1.ProtocolUDP)
						if err != nil {
```

```
	// Finish housekeeping.
	// TODO: these could be made more consistent.
	for _, svcIP := range staleServices.UnsortedList() {
		if err := conntrack.ClearEntriesForIP(proxier.exec, svcIP, v1.ProtocolUDP); err != nil {
			klog.Errorf("Failed to delete stale service IP %s connections, error: %v", svcIP, err)
		}
	}
	proxier.deleteEndpointConnections(endpointUpdateResult.StaleEndpoints)
```